flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/metadata/FlinkRelMdRowCount.scala

```diff
@@ -171,7 +171,8 @@ class FlinkRelMdRowCount private extends MetadataHandler[BuiltInMetadata.RowCoun
     } else {
       val inputRowCnt = mq.getRowCount(input)
       val config = rel.getCluster.getPlanner.getContext.asInstanceOf[FlinkContext].getTableConfig
-      val parallelism = NodeResourceConfig.calOperatorParallelism(inputRowCnt, config.getConf)
+      val parallelism = (inputRowCnt /
+          config.getConf.getLong(PlannerConfigOptions.SQL_OPTIMIZER_ROWS_PER_LOCALAGG) + 1).toInt
```

Comment:
Explain this change?

Reply:
LocalAgg degree was depend on parallelism calculating that happens after plan optimizing, it is unreasonable. So it predicts parallelism itself, that cause no conflict.